odl/discr/discr_ops.py

```diff
@@ -80,16 +81,36 @@ def __init__(self, domain, range):
         super(Resampling, self).__init__(
             domain=domain, range=range, linear=True)
 
+        self.__interp_byaxis = _normalize_interp(interp, domain.ndim)
+
+    @property
+    def interp_byaxis(self):
+        """Tuple of per-axis interpolation schemes."""
+        return self.__interp_byaxis
+
+    @property
+    def interp(self):
+        """Interpolation scheme or tuple of per-axis interpolation schemes."""
+        if _all_interp_equal(self.interp_byaxis):
+            return self.interp_byaxis[0]
+        else:
+            return self.interp_byaxis
+
     def _call(self, x, out=None):
         """Apply resampling operator.
 
         The element ``x`` is resampled using the sampling and interpolation
         operators of the underlying spaces.
         """
-        if out is None:
-            return x.interpolation
-        else:
-            out.sampling(x.interpolation)
+        interpolator = per_axis_interpolator(
+            x, self.domain.grid.coord_vectors, self.interp
+        )
+
+        context = none_context if out is None else writable_array
```

Comment:
The idea is to make the rest of the code unconditional, because it can deal with `out_arr is None` itself. Compare this version to the alternative:
```python
if out is None:
    return point_collocation(
        interpolator, self.range.meshgrid, out=out_arr
    )
else:
    with writable_array(out) as out_arr:
        return point_collocation(
            interpolator, self.range.meshgrid
        )    
```

As a side note, for Python 3.7 and higher, we can move to [`contextlib.nullcontext`](https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext) if we want, but I don't feel like introducing a backports module just for that :-).